src/network/getters/getSponsorshipStats.ts

```diff
@@ -0,0 +1,110 @@
+import moment from 'moment'
+import { ChartPeriod } from '~/shared/components/NetworkChart/NetworkChart'
+import { GetSponsorshipDailyBucketsQuery } from '~/generated/gql/network'
+import { getSponsorshipDailyBuckets } from '~/network/getters/getSponsorshipDailyBuckets'
+import { toBN } from '~/utils/bn'
+import { getSponsorshipTokenInfo } from '~/network/getters/getSponsorshipTokenInfo'
+import { TokenInformation } from '~/marketplace/utils/web3'
+
+export const getSponsorshipStats = async (
+    sponsorshipId: string,
+    selectedPeriod: ChartPeriod,
+    dataSource: string,
+    ignoreToday?: boolean,
+): Promise<{ x: number; y: number }[]> => {
+    const tokenInfo = (await getSponsorshipTokenInfo()) as TokenInformation
+    const todayUTC = moment().utc().startOf('day').format('X')
+    let result: GetSponsorshipDailyBucketsQuery['sponsorshipDailyBuckets']
+    switch (selectedPeriod) {
+        case ChartPeriod.SevenDays:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                7,
+                0,
+                ignoreToday ? todayUTC : undefined,
+            )
+            break
+        case ChartPeriod.OneMonth:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                30,
+                0,
+                ignoreToday ? todayUTC : undefined,
+            )
+            break
+        case ChartPeriod.ThreeMonths:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                90,
+                0,
+                ignoreToday ? todayUTC : undefined,
+            )
+            break
+        case ChartPeriod.OneYear:
+            result = await getSponsorshipDailyBuckets(
+                sponsorshipId,
+                365,
+                0,
+                ignoreToday ? todayUTC : undefined,
+            )
+            break
+        case ChartPeriod.YearToDate:
+            const today = moment()
```

Comment:
`switch` doesn't create a lexical scope for each `case`. In other words, you may end up with conflicting const names. Just FYI.

Doesn't eslint complain about it?

Reply:
good point - even if it doesn't complain - we actually need to initialize this const only once (a day) :)